cpp/net/connection_pool.cc

```diff
@@ -90,12 +229,14 @@ unique_ptr<ConnectionPool::Connection> ConnectionPool::Get(const URL& url) {
 
   if (it == conns_.end() || it->second.empty()) {
     VLOG(1) << "new evhtp_connection for " << key.first << ":" << key.second;
-    unique_ptr<ConnectionPool::Connection> conn(
-        new Connection(base_->HttpConnectionNew(key.first, key.second),
-                       move(key)));
-    evhtp_set_hook(&conn->connection()->hooks, evhtp_hook_on_connection_fini,
+    unique_ptr<ConnectionPool::Connection> conn(new Connection(
+        url.Protocol() == "https"
+            ? base_->HttpsConnectionNew(key.first, key.second, ssl_ctx_.get())
```

Comment:
That would be the best, if a new version of libevhtp would have the fix, but I'm thinking we're likely to need a workaround before then (as in, ASAP).